core/src/main/java/org/apache/druid/data/input/RetryingInputEntity.java

```diff
@@ -32,12 +33,13 @@
   @Override
   default InputStream open() throws IOException
   {
-    return new RetryingInputStream<>(
+    RetryingInputStream<?> retryingInputStream = new RetryingInputStream<>(
         this,
         new RetryingInputEntityOpenFunction(),
         getRetryCondition(),
         RetryUtils.DEFAULT_MAX_TRIES
     );
+    return CompressionUtils.decompress(retryingInputStream, getDecompressionPath());
```

Comment:
nit: thank you for fixing this! I think The javadoc of `readFrom` and `readFromStart` now should mention the returned inputStream shouldn't decompress. I think I'm going to raise a PR for adding some unit tests for this bug and maybe I can update javadoc in my PR unless you want to do it here.